qutip/solver/ode/evolver.py

```diff
@@ -0,0 +1,141 @@
+        self.name = "qutip " + self.options.ode['method']
+
+    def get_state(self, copy=True):
+        state = self._ode_solver.y
+        return self._ode_solver.t, state.copy() if copy else state
+
+    def set_state(self, t, state):
+        self._ode_solver.set_initial_value(state, t)
+
+    def backstep(self, t):
+        self._ode_solver.integrate(t)
+        return self.get_state()
+
+    def step(self, t):
+        """ Evolve to t, must be `set` before. """
+        self._ode_solver.integrate(t)
+        if not self._ode_solver.successful():
+            raise Exception(self._error_msg)
+        return self.get_state()
+
+    def one_step(self, t):
+        """ Evolve to t, must be `set` before. """
+        self._ode_solver.integrate(t, step=True)
+        if not self._ode_solver.successful():
+            raise Exception(self._error_msg)
+        return self.get_state()
+
+
+vernlimits = {
+    "base": True,
+    "backstep": True,
+    "update_args": True,
+    "feedback": True,
+    "time_dependent": True,
+}
+
+
+evolver_collection.add(EvolverVern, methods=['vern7', 'vern9'],
+                       limits=vernlimits, _test=False)
+
+
+class EvolverDiag(Evolver):
+    # -------------------------------------------------------------------------
+    # Solve an ODE for func.
+    # Calculate the required expectation values or invoke callback
+    # function at each time step.
+    #
+    # Diagonalize the Hamiltonian and
+    # This should be used for constant Hamiltonian evolution (sesolve, mcsolve)
+    #
+    description = "Diagonalize a constant system"
+    used_options = []
+
+    def __init__(self, system, options, args, feedback_args):
+        if not system.const:
+            raise ValueError("Hamiltonian system must be constant to use "
+                             "diagonalized method")
+        self.system = system
+        self.options = options
+        self._dt = 0.
+        self._expH = None
+        self.prepare()
+
+    def prepare(self):
+        self.diag, self.U = self.system(0).eigenstates()
+        self.diag = self.diag.reshape((-1,1))
+        self.U = np.hstack([eket.full() for eket in self.U])
+        self.Uinv = np.linalg.inv(self.U)
+        self.name = "qutip diagonalized"
+
+    def step(self, t):
+        """ Evolve to t, must be `set` before. """
+        dt = t - self._t
+        if dt == 0:
+            return self.get_state()
+        elif self._dt != dt:
+            self._expH = np.exp(self.diag * dt)
+            self._dt = dt
+        self._y *= self._expH
+        self._t = t
+        return self.get_state()
+
+    def one_step(self, t):
+        return self.step(t)
+
+    def backstep(self, t):
+        return self.step(t)
+
+    def get_state(self, copy=False):
+        y = self.U @ self._y
+        return self._t, _data.dense.fast_from_numpy(y)
+
+    def set_state(self, t, state0):
+        self._t = t
+        self._y = (self.Uinv @ state0.to_array())
```

Comment:
I think these are self explanatory by still a single line docstring would be good to keep the docs looking nice.